packages/descendants/src/Highlight/README.md

```diff
@@ -0,0 +1,115 @@
+# Highlight Management
+
+A suite of utilities for managing the highlight state of Descendants.
+
+# Usage
+
+```tsx
+/** HighlightDemo.tsx */
+
+const HighlightDemo = () => {
+  // Initialize a new Descendants object
+  const { getDescendants, dispatch } = useInitDescendants<HTMLDivElement>();
+  // Initialize a new highlight state
+  const { highlight, moveHighlight, setHighlight } =
+    useHighlight(getDescendants);
+
+  return (
+    // Init the Descendants context provider
+    <DescendantsProvider
+      context={TestDescendantContext}
+      descendants={getDescendants()}
+      dispatch={dispatch}
+    >
+      {/* Init the Highlight context provider */}
+      <HighlightProvider
+        context={TestHighlightContext}
+        highlight={highlight}
+        setHighlight={setHighlight}
+      >
+        {items.map(item => (
+          <HighlightItem key={item}>{item}</HighlightItem>
+        ))}
+      </HighlightProvider>
+    </DescendantsProvider>
+  );
+};
+```
+
+```tsx
+/** HighlightItem.tsx */
+
+const HighlightItem = ({ children }: PropsWithChildren<{}>) => {
+  const { ref, id } = useDescendant(TestDescendantContext);
+  const { highlight } = useHighlightContext(TestHighlightContext);
+
+  return (
+    <div
+      ref={ref}
+      style={{
+        color: isHighlighted ? 'red' : 'black',
+      }}
+    >
+      {children}
+    </div>
+  );
+};
+```
+
+## `options.onInit`: Setting the initial highlight
+
+To set the initial highlight (or run side-effects once on init), provide an `onInit` callback to the `options` argument of `useHighlight`.
+
+```ts
+const { getDescendants, dispatch } = useInitDescendants<HTMLDivElement>();
+const { highlight, setHighlight } = useHighlight(getDescendants, {
+  // Fired once when the descendants are initialized
+  onInit: () => {
```

Comment:
is this too many renders? can we jsut bake this in?

Reply:
We need to wait for the first render for `descendants` to actually refer to a DOM element, only then can we accurately set the initial highlight